apps/api/src/global.d.ts

```diff
@@ -0,0 +1,13 @@
+// These are the type declarations for the virtual modules
+// used by the `/courses/{id}` and `/instructors/{id}` routes.
+// The reason we have to use inline imports here is because with `import` statements, TypeScript
+// treats this as a normal module rather than an ambient module, and so the module declarations
+// don't actually work when consumed by other TypeScript source files.
+
+declare module "virtual:courses" {
+  declare const courses: Record<string, import("@peterportal-api/types").Course>;
+}
+
+declare module "virtual:instructors" {
+  declare const instructors: Record<string, import("@peterportal-api/types").Instructor>;
+}
```

Comment:
Honestly, it would be great if I could find any official resource that actually describes ambient declaration files adequately, but I can't lol. I think jsdoc formatted like this might be more logical?

 ```suggestion
/**
 * Ambient declaration file for defining "virtual" modules/files. 
 * The file contents are generated dynamically during build time by esbuild.
 * DO NOT add any imports/exports; that converts the file to a regular module and removes the global declarations.
 * @see (link to ambient declaration information that I don't have)
 */


/**
 * Virtual module for caching course information during build time.
 */
declare module "virtual:courses" {
  declare const courses: Record<string, import("@peterportal-api/types").Course>;
}

/**
 * Virtual module for caching instructor information during build time.
 */
declare module "virtual:instructors" {
  declare const instructors: Record<string, import("@peterportal-api/types").Instructor>;
}
```